homeassistant/components/gpsd/__init__.py

```diff
@@ -2,19 +2,54 @@
 
 from __future__ import annotations
 
+from dataclasses import dataclass
+
+from gps3.agps3threaded import AGPS3mechanism
+
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import Platform
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
 from homeassistant.core import HomeAssistant
 
 PLATFORMS: list[Platform] = [Platform.SENSOR]
 
+type GPSDConfigEntry = ConfigEntry[GPSDData]
+
+
+@dataclass
+class GPSDData:
+    """GPSD data type."""
+
+    agps_thread: AGPS3mechanism
+
```

Comment:
If you're only storing one thing inthere, might as well just use that directly

Reply:
Fixed, thanks